src/main/java/edu/hw4/AnimalHelper.java

```diff
@@ -0,0 +1,306 @@
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Task 12: Count animals whose weight is greater than their height.
+     *
+     * @param animals The list of animals to count based on weight exceeding height.
+     * @return The number of animals in the list where weight is greater than height.
+     */
+    public static Integer task12CountAnimalsWeightGreaterThanHeight(List<Animal> animals) {
+        return (int) animals.stream()
+            .filter(animal -> animal.weight() > animal.height())
+            .count();
+    }
+
+    /**
+     * Task 13: Find animals with names consisting of more than two words.
+     *
+     * @param animals The list of animals to search for animals with multi-word names.
+     * @return A list of animals with names consisting of more than two words.
+     */
+    public static List<Animal> task13FindAnimalsWithNamesConsistingOfMoreThanTwoWords(List<Animal> animals) {
+        return animals.stream()
+            .filter(animal -> animal.name().split(" ").length > 2)
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Task 14: Check if there is a dog taller than a specified height.
+     *
+     * @param animals The list of animals to search for a tall dog.
+     * @param k       The minimum height required for a dog to be considered tall.
+     * @return True if there is a dog taller than the specified height; otherwise, false.
+     */
+    public static boolean task14HasDogTallerThanK(List<Animal> animals, int k) {
+        return animals.stream()
+            .anyMatch(animal -> animal.type() == Animal.Type.DOG && animal.height() > k);
+    }
+
+    /**
+     * Task 15: Calculate the sum of weights for each animal type within a specified age range.
+     *
+     * @param animals The list of animals to calculate sum of weights.
+     * @param k       The lower age limit for animals to be considered.
+     * @param l       The upper age limit for animals to be considered.
+     * @return A map associating each animal type with the total sum of weights of animals within the age range.
+     */
+    public static Map<Animal.Type, Integer> task15SumOfWeightsByTypeAndAgeRange(List<Animal> animals, int k, int l) {
+        return animals.stream()
+            .filter(animal -> animal.age() >= k && animal.age() <= l)
+            .collect(Collectors.groupingBy(Animal::type, Collectors.summingInt(Animal::weight)));
+    }
+
+    /**
+     * Task 16: Sort a list of animals by type, sex, and name.
+     *
+     * @param animals The list of animals to be sorted.
+     * @return A new list of animals sorted by type, sex, and name.
+     */
+    public static List<Animal> task16SortAnimalsByTypeSexName(List<Animal> animals) {
+        return animals.stream()
+            .sorted(Comparator.comparing(Animal::type)
+                .thenComparing(Animal::sex)
+                .thenComparing(Animal::name))
+            .collect(Collectors.toList());
```

Comment:
Можно использовать просто
```suggestion
            .toList();
```

Reply:
Поправил